### ISSUE TITLE
Implement XGBoostClassifier

### ISSUE THREAD

**Issue by motiwari** [ThrunGroup/FastForest#72]
We should set this up as a baseline.

It may be valuable to have this as a separate class that inherits `Tree` or `TreeClassifier`.

It should take parameters like the boosting learning rate etc.
